packages/react-native-platform-override/src/Manifest.ts

```diff
@@ -194,93 +102,51 @@ export default class Manifest {
    * Returns the entry corresponding to the given override path, or null if none
    * exists.
    */
-  findOverride(overridePath: string): ManifestData.Entry | null {
-    const idx = this.findOverrideIndex(overridePath);
+  findOverride(overrideName: string): Override | null {
+    const idx = this.findOverrideIndex(overrideName);
     if (idx === -1) {
       return null;
     }
 
-    return _.cloneDeep(this.overrides[idx]);
+    return this.overrides[idx];
   }
 
   /**
    * Updates an override entry to mark it as up-to-date in regards to its
    * current base file.
    */
-  async markUpToDate(overridePath: string) {
-    const entryIdx = this.findOverrideIndex(overridePath);
+  async markUpToDate(overrideName: string, overrideFactory: OverrideFactory) {
+    const entryIdx = this.findOverrideIndex(overrideName);
     if (entryIdx === -1) {
-      throw new Error(`Override at '${overridePath}' does not exist`);
-    }
-
-    const internalEntry = this.overrides[entryIdx];
-    if (internalEntry.type === 'platform') {
-      throw new Error('Cannot mark a platform override as up-to-date');
+      throw new Error(`Override '${overrideName}' does not exist`);
     }
 
-    const newBaseInfo = await this.getOverrideBaseInfo(internalEntry.baseFile);
-    Object.assign(internalEntry, internalEntry, newBaseInfo);
-  }
-
-  /**
-   * Return a copy of the manifest as raw data
-   */
-  getAsData(): ManifestData.Manifest {
-    return {overrides: _.cloneDeep(this.overrides)};
-  }
+    const currentOverride = this.overrides[entryIdx];
+    const upToDateOverride = await currentOverride.createUpdated(
+      overrideFactory,
+    );
 
-  /**
-   * Returns the version of React Native the manifest considers as current
-   */
-  currentVersion(): string {
-    return this.reactRepo.getVersion();
+    // Be careful not to rely on the saved index which may no longer be
+    // accurate after resuming. Modify the original retrieved object instead.
+    Object.assign(currentOverride, upToDateOverride);
   }
 
   /**
-   * Hash content into the form expected in a manifest entry. Exposed for
-   * testing.
+   * Return a serialized representation of the manifest
    */
-  static hashContent(str: string) {
-    const hasher = crypto.createHash('sha1');
-    const normalizedStr = str.replace(/(?<!\r)\n/g, '\r\n');
-    hasher.update(normalizedStr);
-    return hasher.digest('hex');
+  serialize(): Serialized.Manifest {
+    return {
+      overrides: this.overrides
+        .sort((a, b) => a.name().localeCompare(b.name()))
```

Comment:
since the names are identifiers, are we sure we should sort locale-specific

Reply:
Makes sense. Can move from `localCompare` to comparison operators.

Reply:
Actually I'm not sure that I like how casing is handled for pure lexocographic sort. Going to hardcode to English locale to get determinism though.